### PR TITLE
feat: remove domain suffix from generated name

### PR DESCRIPTION
**Pull request by lingsamuel**
APISIX has a limit of 100 characters. Remove the suffix to avoid easily encountering this limit.

### CODE REVIEW

File: pkg/adaptor/xds/v3/route.go [hunk 1]
```diff
@@ -87,6 +87,7 @@ func (adaptor *adaptor) translateVirtualHost(prefix string, vhost *routev3.Virtu
 		}
 		vars = append(vars, queryVars...)
 		name = fmt.Sprintf("%s#%s#%s", name, vhost.GetName(), prefix)
+		name = strings.Replace(name, ".svc.cluster.local", "", -1) // avoid name too long
```
Comment (tokers) [inline pkg/adaptor/xds/v3/route.go:90]: Just use `strings.TrimSuffix`.

Also, this works most of time but will out of work if the cluster domain is not `cluster.local`.

Comment (lingsamuel) [reply pkg/adaptor/xds/v3/route.go:90]: `.svc.cluster.local` is not configurable now
https://github.com/api7/apisix-mesh-agent/blob/309cdb5a16dc06b15965de47939442aacf27fb51/pkg/provisioner/xds/v3/grpc/types.go#L86-L87

Comment (lingsamuel) [reply pkg/adaptor/xds/v3/route.go:90]: `strings.TrimSuffix` is not enough. vhost.Name could be FQDN.

Comment (tokers) [reply pkg/adaptor/xds/v3/route.go:90]: In which case `.svc.cluster.local` would be inside the FQDN but not as the suffix?

**Comment by tokers** [comment]
Conflicts should be resolved firstly @lingsamuel .

File: pkg/adaptor/xds/v3/route.go (hunk 1 above)
Comment (lingsamuel) [reply pkg/adaptor/xds/v3/route.go:90]: Name is concated with schema `"%s#%s#%s"`

Comment (tokers) [reply pkg/adaptor/xds/v3/route.go:107]: OK, I forgot this detail...

**Review by tokers — APPROVED** [review]

**Pull request merged**